scripts/publish-packages.sh

```diff
@@ -1,16 +1,37 @@
+#!/bin/sh
+
+set -eu
+
 SCRIPTS_DIR="$(dirname "$0")"
 cd "${SCRIPTS_DIR}/../packages"
 for d in */ ; do
-    cd $d
+    cd "$d"
 
+    # Get the current package name and version from package.json.
     PACKAGE_NAME=$(node -p "require('./package.json').name")
     PACKAGE_VERSION=$(node -p -e "require('./package.json').version")
     echo "Will check if version ${PACKAGE_VERSION} of ${PACKAGE_NAME} should be published."
 
-    PUBLISHED_VERSION=$(npm view "${PACKAGE_NAME}" version)
+    # Look up the currently published version of the package in the registry.
+    # npm view will fail if the package does not exist. This case needs to be handled
+    # in order to support publishing new packages for the first time.
+    set +e
+    PUBLISHED_VERSION=$(npm --loglevel=error view "${PACKAGE_NAME}" version 2>&1)
```

Comment:
Another way you can write something like this (so you don't have to "remember" to set/unset):
```sh
PUBLISHED_VERSION=$(set +e; npm --loglevel=error view "${PACKAGE_NAME}" version 2>&1)
```
That `set +e` is executed in a subshell, so the setting will not leak into the enclosing environment.

Reply:
With this, the script still exits when `npm view` returns an error because the subshell returns an error.

Reply:
I was able to remove the `set +e` in https://github.com/DataBiosphere/terra-ui/pull/4125/commits/27b02cb7c8162c69cb08c306d94fa6408d2eca10 by moving the `npm view` call into the if statement.